DemoCryptoLib/DemoCryptoLib.cs

```diff
@@ -24,13 +26,14 @@ public static string GeneratePasswordHashUsingSalt(string passwordText, byte[] s
     public static string GeneratePasswordHashUsingSaltOptimized(string passwordText, byte[] salt)
     {
         const int iterate = 10000;
-        using var pbkdf2 = new Rfc2898DeriveBytes(passwordText, salt, iterate);
+        var pbkdf2 = new Rfc2898DeriveBytes(passwordText, salt, iterate);
 
-        byte[] hash = pbkdf2.GetBytes(20);
+        var hash = pbkdf2.GetBytes(20);
```

Comment:
Can you try the methods that operate with spans to avoid this allocation?
https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.rfc2898derivebytes?view=net-7.0

Reply:
Thanks a lot!
The documentation gave me the insight, where to go.
Now new method works ~20% faster and uses 80% less memory.